### PR TITLE
Fix authentication in restapi

### PR DESCRIPTION
**Pull request by sumukhatv**
**- What I did**
Fix authentication issues in restapi server
**- How I did it**
enabling https endpoint and mounting the certificates directory
Also updating the sonic-restapi submodule
**- How to verify it**

**- Description for the changelog**

Enable https endpoint for restapi server

**- A picture of a cute animal (not mandatory but encouraged)**

### CODE REVIEW

File: dockers/docker-sonic-restapi/start.sh [hunk 1]
```diff
@@ -7,4 +7,14 @@ rm -f /var/run/rsyslogd.pid
...
+SERVER_CERT=/etc/sonic/certificates/restapiserver.crt
+SERVER_KEY=/etc/sonic/certificates/restapiserver.key
+CLIENT_CERT=/etc/sonic/certificates/restapiclient.crt
+
+while [ : ]
+do
+    if [[ -f $SERVER_CERT && -f $SERVER_KEY && -f $CLIENT_CERT ]]; then
+        supervisorctl start restapi
+    fi
+    sleep 5s
```
Comment (lguohan) [inline dockers/docker-sonic-restapi/start.sh:19]: suggest to increase the 60 seconds, and should generate a syslog info level to waiting for certs to start restapi server ....

File: dockers/docker-sonic-restapi/supervisord.conf [hunk 2]
```diff
@@ -3,10 +3,26 @@ logfile_maxbytes=1MB
...
+[program:start.sh]
+command=/usr/bin/start.sh
 priority=1
 autostart=true
+autorestart=true
+stdout_logfile=syslog
+stderr_logfile=syslog
+
+[program:restapi]
+command=/usr/sbin/go-server-server -enablehttps=true -clientcert=/etc/sonic/certificates/restapiclient.crt -servercert=/etc/sonic/certificates/restapiserver.crt -serverkey=/etc/sonic/certificates/restapiserver.key -loglevel=trace
```
Comment (lguohan) [inline dockers/docker-sonic-restapi/supervisord.conf:15]: should look for this as example. 

https://github.com/Azure/sonic-buildimage/blob/master/dockers/docker-sonic-telemetry/telemetry.sh

we should read the certs form config db.

File: dockers/docker-sonic-restapi/supervisord.conf [hunk 3]
```diff
@@ -3,10 +3,26 @@ logfile_maxbytes=1MB
 logfile_backups=2
 nodaemon=true
 
-[program:restapi]
-command=/usr/sbin/go-server-server -loglevel trace
+[program:start.sh]
```
Comment (lguohan) [inline dockers/docker-sonic-restapi/supervisord.conf:6]: please also add critical process monitor. 
https://github.com/Azure/sonic-buildimage/blob/master/dockers/docker-sonic-telemetry/supervisord.conf#L7

File: dockers/docker-sonic-restapi/supervisord.conf [hunk 4]
```diff
@@ -3,10 +3,26 @@ logfile_maxbytes=1MB
...
 autorestart=false
-stdout_logfile=/tmp/rest-api.out.log
-stderr_logfile=/tmp/rest-api.err.log
+stdout_logfile=syslog
+stderr_logfile=syslog
+
+[program:rsyslogd]
+command=/usr/sbin/rsyslogd -n
+priority=2
+autostart=false
```
Comment (lguohan) [inline dockers/docker-sonic-restapi/supervisord.conf:25]: autostart should be true.

File: rules/docker-restapi.mk [hunk 5]
```diff
@@ -19,4 +19,5 @@ endif
 $(DOCKER_RESTAPI)_CONTAINER_NAME = restapi
 $(DOCKER_RESTAPI)_RUN_OPT += --cap-add NET_ADMIN --privileged -t
 $(DOCKER_RESTAPI)_RUN_OPT += -v /var/run/redis/redis.sock:/var/run/redis/redis.sock
-$(DOCKER_RESTAPI)_RUN_OPT += -p=8090:8090/tcp
+$(DOCKER_RESTAPI)_RUN_OPT += -v /etc/sonic/certificates:/etc/sonic/certificates:ro
+$(DOCKER_RESTAPI)_RUN_OPT += -p=8081:8081/tcp
```
Comment (lguohan) [inline rules/docker-restapi.mk:23]: why change port number?

File: dockers/docker-sonic-restapi/supervisord.conf (hunk 4 above)
Comment (sumukhatv) [reply dockers/docker-sonic-restapi/supervisord.conf:25]: This is being started from the start script.

---
In reply to: [404614583](https://github.com/Azure/sonic-buildimage/pull/4383#discussion_r404614583) [](ancestors = 404614583)

File: rules/docker-restapi.mk (hunk 5 above)
Comment (sumukhatv) [reply rules/docker-restapi.mk:23]: 8081 is the https endpoint.
8090 is for http endpoint

---
In reply to: [404614708](https://github.com/Azure/sonic-buildimage/pull/4383#discussion_r404614708) [](ancestors = 404614708)

File: dockers/docker-sonic-restapi/supervisord.conf (hunk 4 above)
Comment (jleveque) [reply dockers/docker-sonic-restapi/supervisord.conf:25]: This is correct.

File: dockers/docker-sonic-restapi/supervisord.conf [hunk 6]
```diff
@@ -3,10 +3,26 @@ logfile_maxbytes=1MB
...
+autorestart=unexpected
+stdout_logfile=syslog
+stderr_logfile=syslog
```
Comment (jleveque) [inline dockers/docker-sonic-restapi/supervisord.conf:28]: Add newline at EOF

File: dockers/docker-sonic-restapi/start.sh [hunk 7]
```diff
@@ -7,4 +7,14 @@ rm -f /var/run/rsyslogd.pid
...
+done
```
Comment (jleveque) [inline dockers/docker-sonic-restapi/start.sh:20]: Add newline at EOF

File: rules/docker-restapi.mk (hunk 5 above)
Comment (lguohan) [reply rules/docker-restapi.mk:23]: should we then bind both?

**Comment by lguohan** [comment]
retest vsimage please

File: rules/docker-restapi.mk (hunk 5 above)
Comment (prsunny) [reply rules/docker-restapi.mk:23]: Yes, i think we should bind both

Comment (sumukhatv) [reply rules/docker-restapi.mk:23]: If we don't want to bring-up the http endpoint at all, why do we need to bind the 8090 port?

Comment (prsunny) [reply rules/docker-restapi.mk:24]: it might be required for testing, we can probably remove it later

Comment (sumukhatv) [reply rules/docker-restapi.mk:24]: We test the restapi server using the restapi test container image generated by https://github.com/Azure/sonic-restapi/blob/master/build.sh and the configuration file https://github.com/Azure/sonic-restapi/blob/master/supervisor/rest_api_test.conf which will enable the http endpoint. Therefore I don't think we need to keep the port open here.

Comment (sumukhatv) [reply rules/docker-restapi.mk:24]: I have added port 8090 also now

---
In reply to: [406905231](https://github.com/Azure/sonic-buildimage/pull/4383#discussion_r406905231) [](ancestors = 406905231)

**Comment by sumukhatv** [comment]
retest vsimage

**Comment by sumukhatv** [comment]
retest vsimage please

**Pull request merged**